### ISSUE TITLE
Error when running cibersort on ALL patients with the Custom Signature Matrix

### ISSUE THREAD

**Issue by vanleiko** [ShahriyariLab/TumorDecon#2]
Hi,

I'm following the full_tutorial and when I try to run this part of the code using the **custom signature matrix** created in this tutorial:

```
ciber_freqs = td.tumor_deconvolve(rna, 'cibersort',  patient_IDs='ALL', 
                                  cell_signatures=sig2, args={'nu':'best', 'scaling':'minmax'})

print("CIBERSORT Results:")

print(ciber_freqs)
``` 

**I get this ValueError:**

```
ValueError                                Traceback (most recent call last)
<ipython-input-18-f225c396ea4b> in <module>()
      1 ciber_freqs = td.tumor_deconvolve(rna, 'cibersort',  patient_IDs='ALL', 
----> 2                                   cell_signatures=sig2, args={'nu':'best', 'scaling':'minmax'})
      3 print("CIBERSORT Results:")
      4 print(ciber_freqs)

5 frames
/usr/local/lib/python3.7/dist-packages/sklearn/utils/validation.py in check_consistent_length(*arrays)
    210     if len(uniques) > 1:
    211         raise ValueError("Found input variables with inconsistent numbers of"
--> 212                          " samples: %r" % [int(l) for l in lengths])
    213 
    214 

ValueError: Found input variables with inconsistent numbers of samples: [522, 524]
```

**But when I use the default signature matrix file (LM22), it works.** 
Can you tell me what I'm doing wrong when using the custom signature matrix?

Thanks!

**Comment by tyaoi** [comment]
Hi,

When I replace the sample data to my own RNA-seq data in the tutorial.py and run, I get the same ValueError in both linear models.

Rank-Based Methods, however, return the results.

I can't understand what the message means, and resolve the errors.
Would you tell me any advises or comment?

Thanks,

**Comment by raronow** [comment]
Thank you for catching this! The line to remove any duplicated genes (rows) in the RNA sequence data was commented out from the read_rna_data() function in a recent commit - this resulted in inconsistent sizes between the signature matrix and mixture data whenever 1 of the 2 files had multiple rows for a given gene. This has been patched in the latest release (1.1.1)
